### DISCUSSION TITLE
Where to place the QrBill Section

### DISCUSSION THREAD

**Discussion by staatzstreich** [schoero/SwissQRBill#400] [General]
Hello...

i am generating Invoices with react-pdf and have the need to deliver a Swiss Qr Bill... My Invoices are Multipage and with a fixed Header and Footer.

Should I add a blank A4 sized page with the Qr Bill at the bottom or is it ok to include only the A6/5 sized Version?

I ask because, if you open the A6/5 sized version generated PDF like so
```
import { PDF } from "swissqrbill/pdf";

const data = {
  currency: "CHF",
  amount: 1199.95,
  reference: "210000000003139471430009017",
  creditor: {
    name: "Robert Schneider AG",
    address: "Rue du Lac",
    buildingNumber: "1268",
    zip: 2501,
    city: "Biel",
    account: "CH4431999123000889012",
    country: "CH"
  },
  debtor: {
    name: "Pia-Maria Rutschmann-Schnyder",
    address: "Grosse Marktgasse",
    buildingNumber: "28",
    zip: 9400,
    city: "Rorschach",
    country: "CH"
  }
};

const pdf = new PDF(data, "qrbill.pdf", () => {
  console.log("PDF has been successfully created.");
});
```

on MacOS with the Preview-App it looks good but with Printing Preview the PDF is "Auto rotated" and Scaled. It is placed center of a A4 Page.

Is this ok with Swiss Banking Institutes?

For me as an no Swiss Citizen this is very confusing. Can you help with clarification ?

Thanks and Best,
Michael

**Comment by schoero** [comment]
I would recommend to change the size to `"A4"` to be on the safe side.  Any scaling or even rotating the printer does would technically violate the specifications, although I'm sure the banks would still accept such invoices.

You don't have to add a blank page manually, you can just change the size with the [`size` option](https://github.com/schoero/SwissQRBill/blob/v3.2.2/doc/api.md#options). It will then automatically add a new (blank) page if there is not enough space left.

```ts
const pdf = new PDF(data, "qrbill.pdf", { "size": "A4" }, () => {
  console.log("PDF has been successfully created.");
});
```

You may need to use the [Switching to previous pages](https://pdfkit.org/docs/getting_started.html#switching_to_previous_pages) feature from PDFKit to create the Header and Footer. That way you can check if you are on the last page and omit the Footer so that it won't interfere with the payment part.

```ts
const pdf = new PDF(data, "qrbill.pdf", { "autoGenerate": false, "size": "A4" });
// Create your invoice
pdf.addQRBill();

pdf.on("beforeEnd", () => {
  const range = pdf.bufferedPageRange();
  for(let i = range.start, end = range.start + range.count; i < end; i++){
    pdf.switchToPage(i);
    if(i !== end){
      // Render Header and Footer
    }
  }
});

pdf.end();
```

**Reply by staatzstreich** [reply]
Hello Roger, 

thank you very much for the quick and detailed answer. That helps a lot.

best,
Michael
